metadata-ingestion/src/datahub/cli/ingest_cli.py

```diff
@@ -0,0 +1,284 @@
+
+    config_file = pathlib.Path(config)
+    pipeline_config = load_config_file(config_file)
+
+    try:
+        logger.debug(f"Using config: {pipeline_config}")
+        pipeline = Pipeline.create(pipeline_config)
+    except ValidationError as e:
+        click.echo(e, err=True)
+        sys.exit(1)
+
+    logger.info("Starting metadata ingestion")
+    pipeline.run()
+    logger.info("Finished metadata ingestion")
+    ret = pipeline.pretty_print_summary()
+    sys.exit(ret)
+
+
+def get_runs_url(gms_host: str) -> str:
+    return f"{gms_host}/runs?action=rollback"
+
+
+def parse_restli_response(response):
+    response_json = response.json()
+
+    if not isinstance(response_json, dict):
+        click.echo(f"Received error, please check your {CONDENSED_DATAHUB_CONFIG_PATH}")
+        click.echo()
+        click.echo(response_json)
+        exit()
+
+    rows = response_json.get("value")
+    if not isinstance(rows, list):
+        click.echo(f"Received error, please check your {CONDENSED_DATAHUB_CONFIG_PATH}")
+        click.echo()
+        click.echo(response_json)
+        exit()
+
+    return rows
+
+
+def parse_run_restli_response(response):
+    response_json = response.json()
+
+    if not isinstance(response_json, dict):
+        click.echo(f"Received error, please check your {CONDENSED_DATAHUB_CONFIG_PATH}")
+        click.echo()
+        click.echo(response_json)
+        exit()
+
+    summary = response_json.get("value")
+    if not isinstance(summary, dict):
+        click.echo(f"Received error, please check your {CONDENSED_DATAHUB_CONFIG_PATH}")
+        click.echo()
+        click.echo(response_json)
+        exit()
+
+    return summary
+
+
+def print_datahub_env_format_guide():
+    click.secho(
+        f"datahub config ({CONDENSED_DATAHUB_CONFIG_PATH}) is malformed.", bold=True
+    )
+    click.echo("see expected format below...")
+    click.echo()
+    click.echo("gms:")
+    click.echo("  server: <gms host>")
+    click.echo("  token: <optional gms token>")
+
+
+def get_session_and_host():
+    session = requests.Session()
+
+    gms_host = "http://localhost:8080"
+    gms_token = None
+    if not os.path.isfile(DATAHUB_CONFIG_PATH):
+        click.secho(
+            f"No {CONDENSED_DATAHUB_CONFIG_PATH} file found, generating one for you...",
+            bold=True,
+        )
+
+        with open(DATAHUB_CONFIG_PATH, "w+") as outfile:
+            yaml.dump(DEFAULT_DATAHUB_CONFIG, outfile, default_flow_style=False)
+
+    with open(DATAHUB_CONFIG_PATH, "r") as stream:
+        try:
+            config = yaml.safe_load(stream)
+            if not isinstance(config, dict):
+                print_datahub_env_format_guide()
+                exit()
+
+            gms_config = config.get("gms")
+            if not isinstance(gms_config, dict) or gms_config.get("server") is None:
+                print_datahub_env_format_guide()
+                exit()
```

Comment:
We should use pydantic for all of this validation logic

Reply:
Cool- will look into that!

Reply:
added 👍